tests/instruments/keysight/test_keysight81160A_with_device.py

```diff
@@ -0,0 +1,508 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2025 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+
+import pytest
+
+from pymeasure.instruments.keysight.keysight81160A import (
+    AMPLITUDE_RANGE,
+    BURST_NCYCLES,
+    BURST_PERIOD,
+    FREQUENCY_RANGE,
+    FREQUENCY_SIN_RANGE,
+    IMPEDANCE_RANGE,
+    LIMIT_HIGH_RANGE,
+    LIMIT_LOW_RANGE,
+    OFFSET_RANGE,
+    PULSE_DUTYCYCLE_RANGE,
+    PULSE_PERIOD_RANGE,
+    PULSE_TRANSITION_RANGE,
+    PULSE_WIDTH_RANGE,
+    SQUARE_DUTYCYCLE_RANGE,
+    TRIGGER_COUNT,
+    TRIGGER_MODES,
+    VOLTAGE_HIGH_RANGE,
+    VOLTAGE_LOW_RANGE,
+    WF_SHAPES,
+    Keysight81160A,
+)
+
+pytest.skip("Only works with connected hardware", allow_module_level=True)
```

Comment:
Not needed, as the `connected_device_address` fixture already skips the test unless you call it explicitly with and with a device address.
With that line, you will never run this test file!

See https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/tests.html#device-tests